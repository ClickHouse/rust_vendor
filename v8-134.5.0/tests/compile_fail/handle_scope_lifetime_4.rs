// Copyright 2019-2020 the Deno authors. All rights reserved. MIT license.

pub fn main() {
  let mut isolate = v8::Isolate::new(mock());
  let mut scope1 = v8::HandleScope::new(&mut isolate);

  let mut _scope3 = {
    let mut scope2 = v8::HandleScope::new(&mut scope1);
    v8::EscapableHandleScope::new(&mut scope2)
  };
}

fn mock<T>() -> T {
  unimplemented!()
}
