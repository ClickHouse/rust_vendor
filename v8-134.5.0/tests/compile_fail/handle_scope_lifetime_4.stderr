error[E0597]: `scope2` does not live long enough
  --> tests/compile_fail/handle_scope_lifetime_4.rs:9:35
   |
7  |   let mut _scope3 = {
   |       ----------- borrow later stored here
8  |     let mut scope2 = v8::HandleScope::new(&mut scope1);
   |         ---------- binding `scope2` declared here
9  |     v8::EscapableHandleScope::new(&mut scope2)
   |                                   ^^^^^^^^^^^ borrowed value does not live long enough
10 |   };
   |   - `scope2` dropped here while still borrowed
